app/views/play/VoyagerUnitMap.vue

```diff
@@ -0,0 +1,213 @@
+        const classroom = this.classroom = await api.classrooms.get({classroomID: classroomId})
+        const classroomCourseLevels =  _.find(classroom.courses, {_id: courseId}).levels
+
+        // get levels data for the levels in the classroom snapshot
+        const classroomCourseLevelsData = await api.classrooms.getCourseLevels({classroomID: classroomId, courseID: courseId})
+
+        for (let level of classroomCourseLevels) {
+          this.classroomLevelMap[level.original] = level
+        }
+        
+        let courseLevelsData = {}
+        for (let level of classroomCourseLevelsData) {
+          let original = level.original
+          if (existingCampaignLevels[original]) {
+            courseLevelsData[original] = existingCampaignLevels[original]
+          }
+          else {
+            // a level which has been removed from the campaign but is saved in the course
+            courseLevelsData[original] = level
+          }
+          // carry over position, nextLevels, first property stored in classroom course, if there are any
+          if (this.classroomLevelMap[original].position)
+            courseLevelsData[original].position = this.classroomLevelMap[original].position
+          if (this.classroomLevelMap[original].nextLevels)
+            courseLevelsData[original].nextLevels = this.classroomLevelMap[original].nextLevels
+          if (this.classroomLevelMap[original].first)
+            courseLevelsData[original].first = this.classroomLevelMap[original].first
+        }
+        // TODO: what if level was removed from campaign, and classroom level map doesnt contain positions, nextLevels, and first? Is it a valid scenario?
+        // If yes, then use the logic from CampaignView for setting the missing positions using the prev and next level positions
+        // Not sure how to set the .first and .nextLevels in that case.
+
+        return courseLevelsData
+      }
+      catch (err) {
+        return Promise.reject(err)
+      }
+    },
+    createLevelStatusMap() {
+      // Remove the level sessions for the levels played in another language - for the classroom version of unit map
+      // TODO: What if played in same language in another classroom
+      // TODO: What should be done for home version (i.e. played a level in another campaign)
+      if(this.classroomLevelMap && this.classroom) {
+        for (let session of this.levelSessions) {
+          const classroomLevel = this.classroomLevelMap[session.level.original]
+          if (!classroomLevel)
+            continue
+          const expectedLanguage = classroomLevel.primerLanguage || this.classroom.aceConfig.language
+          if (session.codeLanguage != expectedLanguage) {
+            this.levelSessions.splice(this.levelSessions.indexOf(session),1)
+          } 
+        }
+      }
+      this.levelStatusMap = utils.getLevelStatusMap(this.levelSessions)
+    },
+    determineNextLevel() { // set .next and .locked for this.levels
```

Comment:
I'm uncomfortable with complex logic in the Vue components. Should logic be written in a helper or related file and imported.
Testing logic + view rendering compounds complexity.
If we separate out logic we  can test that separate, and then test the vue component more easily by passing in the correct props. 
I do agree that some logic can't be avoided. But these quickly complicated methods like `determineNextLevel` feel like they should stand alone and be tested alone.

Reply:
That's a good idea, although that would mean moving all the logic from the `methods` section to some helper/utility file. I am not sure when to do that, vs have a method in the vue file itself.
I believe it's about how we want to set the best practices for vue going forward, and we can bring up this discussion in the vue best practices doc(if it doesnt include it already).

Although I like the idea, I am keeping as it is, for now, and will update later based on the vue best practices we establish.

Reply:
Also, on that note, I tried to have some common pieces like  `getLevelStatusMap`, and `findNextLevels` in a common utility file called `voyagerUtils.js`.